src/bitops.c

```diff
@@ -764,14 +991,95 @@ void bitopCommand(client *c) {
                     j += sizeof(unsigned long)*4;
                     minlen -= sizeof(unsigned long)*4;
                 }
+            } else if (op == BITOP_DIFF || op == BITOP_DIFF1 || op == BITOP_ANDOR) {
+                size_t processed = 0;
+                while(minlen >= sizeof(unsigned long)*4) {
+                    for (i = 1; i < numkeys; i++) {
+                        lres[0] |= lp[i][k+0];
+                        lres[1] |= lp[i][k+1];
+                        lres[2] |= lp[i][k+2];
+                        lres[3] |= lp[i][k+3];
+                    }
+                    k+=4;
+                    lres+=4;
+                    j += sizeof(unsigned long)*4;
+                    minlen -= sizeof(unsigned long)*4;
+                    processed += sizeof(unsigned long)*4;
+                }
+
+                lres = (unsigned long*) res;
+                unsigned long *first_key = (unsigned long*)src[0];
```

Comment:
src allows non-existent strings. what if src[0] is NULL?

Reply:
Took me a bit of time to reprocess the code, as it looks like obvious mistake, but it's not.
command is in the form `BITOP <OP> <DEST_KEY> <SRC_KEY_1> [... <SRC_KEY_N>]`

The `src` array stores only the source keys - if any of them is empty/NULL then `minlen` is 0 and we don't enter in the fast path nor in AVX path. Each such NULL source key is treated as if it was all zeroes in the regular path.

The destination key on the other hand is saved in `targetkey` and it doesn't matter if it exists or not - it's overwritten with the result of the operation.